docs/jax-101/05.1-pytrees.md

```diff
@@ -255,18 +279,20 @@ def unflatten_MyContainer(
   """Converts aux data and the flat contents into a MyContainer."""
   return MyContainer(aux_data, *flat_contents)
 
-jax.tree_util.register_pytree_node(
-    MyContainer, flatten_MyContainer, unflatten_MyContainer)
+jax.tree_util.register_pytree_with_keys(
+    MyContainer, flatten_with_keys_MyContainer, unflatten_MyContainer)
 
 jax.tree_util.tree_leaves([
     MyContainer('Alice', 1, 2, 3),
     MyContainer('Bob', 4, 5, 6)
 ])
 ```
 
+Note that registering both keys and values is a new feature, and you may see legacy pytrees that only has values registered (aka. registered via `register_pytree_node`). They are free to use most `tree_util` utilities except the `.*_with_path` APIs.
```

Comment:
Saying "new feature" is a bit dangerous in docs because what if we leave this sentence in for a few years, and "new" no longer really means "new" :P

I think we should keep both variants documented, and tweak the language not to talk about the non-key path being "legacy" but rather the key-ful path being an "extended API".

Reply:
Done.